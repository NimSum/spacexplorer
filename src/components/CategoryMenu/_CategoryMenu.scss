div.menu-container {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 10;
  margin: 10px;
  font-size: 1.1rem;
  border-radius: 15px;
  background-color: $primary-dark;
  padding: 5px 10px;
  text-align: center;
  min-width: 170px;
  

  nav {
    display: flex;
    flex-direction: column;
    color: $secondary-light;
    padding-bottom: 10px;

    .is-active {
      background-color: $secondary-dark;
      color: $secondary-dark;
    }
    
    span {
      display: block;
      transition: all 100ms ease-in-out;

      &:hover {
        color: $primary-light;
        transform: scale(1.2);        
      }
    }
  }

  p {
    color: $secondary-light
  }

  p.hovered {
    padding-bottom: 5px;
    border-bottom: 1px solid $primary-light;
    color: $primary-light;
  }

}