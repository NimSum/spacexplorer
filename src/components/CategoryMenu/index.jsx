import React, { Component } from 'react'
import { NavLink, Link } from 'react-router-dom';

export class CategoryMenu extends Component {
  constructor() {
    super();
    this.state = {
      showMenu: false
    }
  }

  generateMenu() {
    const linkStyling = { 
      textDecoration: 'none',
      color: '#FFEAFE',
      padding: '5px',
      borderBottom: '1px solid #9EA9F0'
    }
    return (
      <nav>
        < NavLink 
          style={ linkStyling } 
          to='/about/rockets'
          activeClassName='is-active'>
            <span>Rockets</span>
        </NavLink>
        < NavLink 
          style={ linkStyling } 
          to='/about/space_stations'
          activeClassName='is-active'>
            <span>Space Stations</span>
        </NavLink>
        < NavLink 
          style={ linkStyling } 
          to='/about/orbiters'
          activeClassName='is-active'>
            <span>Orbiters</span>
        </NavLink>
        < NavLink 
          style={ linkStyling } 
          to='/about/astronauts'
          activeClassName='is-active'>
            <span>Astronauts</span>
        </NavLink>
        < NavLink 
          style={ linkStyling } 
          to='/about/launch_providers'
          activeClassName='is-active'>
            <span>Launch Providers</span>
        </NavLink>
        < Link 
          style={ linkStyling } 
          to='/'>
          <span>HOME</span>
        </Link>
      </nav>
    )
  }

  showMenu = (bool) => {
    this.setState({ showMenu: bool })
  }

  render() {
    return (
      <div 
        className='menu-container'
        onMouseEnter={ () => this.showMenu(true) }
        onMouseLeave={ () => this.showMenu(false) }
        >
        <p className={ this.state.showMenu ? 'hovered' : undefined }>Explore Space </p>
        { this.state.showMenu && this.generateMenu()}
      </div>
    )
  }
}

export default CategoryMenu;
