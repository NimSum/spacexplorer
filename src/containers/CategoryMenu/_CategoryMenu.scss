div.menu-container {
  nav {
    display: flex;
    flex-direction: column;
  }

  position: absolute;
  top: 0;
  right: 0;

}