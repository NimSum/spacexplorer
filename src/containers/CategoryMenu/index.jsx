import React, { Component } from 'react'
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

export class CategoryMenu extends Component {
  constructor() {
    super();
    this.state = {
      showMenu: false
    }
  }

  generateMenu() {
    return (
      <nav>
        < NavLink to='/about/rockets'>
          Rockets
        </NavLink>
        < NavLink to='/about/space_stations'>
          Space Stations
        </NavLink>
        < NavLink to='/about/orbiters'>
          Orbiters
        </NavLink>
        < NavLink to='/about/astronauts'>
          Astronauts
        </NavLink>
        < NavLink to='/about/launch_providers'>
          Launch Providers
        </NavLink>
        < NavLink to='/'>
          Upcoming Launches
        </NavLink>
      </nav>
    )
  }

  showMenu = (bool) => {
    this.setState({ showMenu: bool })
  }

  render() {
    return (
      <div 
        className='menu-container'
        onMouseEnter={ () => this.showMenu(true) }
        onMouseLeave={ () => this.showMenu(false) }>
        { this.state.showMenu 
            ? this.generateMenu()
            : <p>Xplore Space</p>
        }
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
})

export default connect(null, mapDispatchToProps)(CategoryMenu);
