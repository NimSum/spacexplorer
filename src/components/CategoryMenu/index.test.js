import React, { Component } from 'react'
import { shallow } from 'enzyme';
import CategoryMenu from './index';

describe('CategoryMenu', () => {
  const defaultState = {
    showMenu: false
  }

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< CategoryMenu />)
  })
  
  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot with menu shown', () => {
    wrapper.setState({ showMenu: true });
    expect(wrapper).toMatchSnapshot();
  })

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  })

  it('should set state showMenu based on param', () => {
    wrapper.instance().showMenu(true);
    expect(wrapper.state().showMenu).toEqual(true);
    wrapper.instance().showMenu(false);
    expect(wrapper.state().showMenu).toEqual(false);
  })

})