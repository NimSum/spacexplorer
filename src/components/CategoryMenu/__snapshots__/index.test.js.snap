// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryMenu should match component snapshot 1`] = `
<div
  className="menu-container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p>
    Explore Space 
  </p>
</div>
`;

exports[`CategoryMenu should match snapshot with menu shown 1`] = `
<div
  className="menu-container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p
    className="hovered"
  >
    Explore Space 
  </p>
  <nav>
    <NavLink
      activeClassName="is-active"
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/about/rockets"
    >
      <span>
        Rockets
      </span>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/about/astronauts"
    >
      <span>
        Astronauts
      </span>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/about/space_stations"
    >
      <span>
        Space Stations
      </span>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/about/orbiters"
    >
      <span>
        Orbiters
      </span>
    </NavLink>
    <NavLink
      activeClassName="is-active"
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/about/launch_providers"
    >
      <span>
        Launch Providers
      </span>
    </NavLink>
    <Link
      style={
        Object {
          "borderBottom": "1px solid #9EA9F0",
          "color": "#FFEAFE",
          "padding": "5px",
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <span>
        HOME
      </span>
    </Link>
  </nav>
</div>
`;
